build.sh

```diff
@@ -49,7 +49,7 @@ run()
 	ret=$?
 
 	if [ $ret -ne 0 ]; then
-		echo "ERROR: '$@' failed, exit code: $ret" >&2
+		echo "ERROR: '$*' failed, exit code: $ret" >&2
```

Comment:
'$@' works well on dash. Is there problem on busybox sh or any other shell?

Reply:
It's mixing arrays and strings, here's a good resource https://www.shellcheck.net/wiki/SC2145

Reply:
POSIX compliant shell syntax, which I use for a reason (i.e. working on dash, which is Debian default or busybox sh which is on many embedded systems) does not have arrays. This is artificial problem. While shellcheck is great to found hidden bugs in complex shell scripts, here is misleading.